sdk/storage/azure-storage-common/inc/azure/storage/common/crypt.hpp

```diff
@@ -7,32 +7,21 @@
 #include <string>
 #include <vector>
 
+#include "azure/core/cryptography/hash.hpp"
 #include <azure/core/base64.hpp>
 
 namespace Azure { namespace Storage {
 
-  class Crc64 {
+  class Crc64Hash : public Azure::Core::Cryptography::Hash {
```

Comment:
I think Jinming has a reason why this class will NOT derive from Core's Hash.

Reply:
>  If it's just because it needs Concatenate but it ALSO works/is useful without Concatenate, then deriving seems good to me.

I think that was his reasoning, but having a concrete class add additional specific APIs is fine, from https://github.com/Azure/azure-sdk-for-cpp/pull/1632#issuecomment-775643215, unless there are other reasons.
> From @Jinming-Hu:
> Plus, I'm not going to make Crc64 class inherit from this hash base. As Crc64 supports concatenate operation, while your base class doesn't.

Reply:
At least one benefit for deriving like this, is that we get consistent and well-defined/documented behavior for the edge cases for free such as what happens when you call `Final()` multiple times or `Append()` after a call to `Final()`. The base class enforces the right behavior.